view/frontend/web/insights/add-to-cart-mixin.js

```diff
@@ -0,0 +1,51 @@
+define(['jquery'], function ($) {
+    'use strict';
+
+    var algoliaAddToCartMixin = {
+
+        submitForm: function (form) {
+            if (window.algoliaConfig
+                && algoliaConfig.ccAnalytics.enabled
+                && algoliaConfig.ccAnalytics.conversionAnalyticsMode == 'place_order'
+            ) {
+                this._setQueryIdToForm(form)
+            }
+
+            this._super(form);
+        },
+
+        _setQueryIdToForm: function(form) {
+            if (form.find('button[type="submit"]').length
+                && form.find('button[type="submit"]').data('queryid')) {
+                var queryID = form.find('button[type="submit"]').data('queryid');
+            }
+
+            var queryID = queryID || this._getQueryParamFromCurrentUrl('queryID');
+            if (queryID.length == 0) {
+                return;
+            }
+
+            if (form.find('input[name="queryid"]').length == 0) {
+                form.prepend('<input type="hidden" name="queryID" />');
+            }
+
+            form.find('input[name="queryID"]').val(queryID);
+        },
+
+        _getQueryParamFromCurrentUrl: function(queryParamName) {
```

Comment:
Maybe this function could be called something more generic as it seems to be able to handle more than just eh queryID?
Something like `parseURL(window.location.href)`.